deploy_tools/generate_compendium.py

```diff
@@ -165,7 +165,7 @@ def slug(
         if add_hash:
             # Create a hash of the entire CompendiumEntry and add its first few characters
             # to the slug.
-            data = json.dumps(self.to_json()).encode("utf-8")
+            data = json.dumps(self.to_json(), indent=None, sort_keys=True, separators=(',',':')).encode("utf-8")
```

Comment:
```suggestion
            uid_keys = ("title", "date", "publisher", "authors")
            uid_dict = {k: v for k, v in self.to_json().items() if k in uid_keys}
            data = json.dumps(uid_dict, indent=None, sort_keys=True, separators=(',',':')).encode("utf-8")
```

This should ensure that the hash only changes when the author, title, publisher, or publication date changes, which will probably only occur when we need to go back to fix a typo. I assume that these four fields are enough to be unique unless there are duplicates in Zotero, but that's an assumption we can revisit later.

Reply:
i've pushed an update that includes this change.

Reply:
Thanks!